units/src/locktime/absolute.rs

```diff
@@ -9,8 +9,12 @@ use internals::write_err;
 #[cfg(feature = "alloc")]
 use crate::parse;
 use crate::parse::ParseIntError;
-#[cfg(feature = "alloc")]
-use crate::prelude::{Box, String};
+
+#[cfg(all(feature = "alloc", not(feature = "std")))]
+use alloc::{string::String, boxed::Box};
+
+#[cfg(feature = "std")]
+use std::{string::String, boxed::Box};
```

Comment:
All this can just become:

```rust
#[cfg(feature = "alloc")]
use alloc::{boxed::Box, string::String};
```

Notes:

- Its ok to use `alloc` types even in `std` bulid - they are the same times as those re-exported from `std`.
- These lines should go above the import of `core::fmt` (`alloc` and `core` imports go together.
- Its ok to violate the use statement style when feature gating

Reply:
Thanks, I wasn't sure if even the `#[cfg(feature = "alloc")]` was necessary, but have left it in as you suggest.